provider/pdns.go

```diff
@@ -345,7 +360,23 @@ func (p *PDNSProvider) ConvertEndpointsToZones(eps []*endpoint.Endpoint, changet
 
 	}
 
-	// If we still have some endpoints left, it means we couldn't find a matching zone for them
+	// residualZones is unsorted by name length like its counterpart
+	// since we only care to remove endpoints that do not match domain filter
+	for _, zone := range residualZones {
+		for i := 0; i < len(endpoints); {
+			ep := endpoints[i]
+			dnsname := ensureTrailingDot(ep.DNSName)
+			if dnsname == zone.Name || strings.HasSuffix(dnsname, "."+zone.Name) {
+				// "pop" endpoint if it's matched to a residual zone... essentially a no-op
+				log.Debugf("Ignoring Endpoint because it was matched to a zone that was not specified within Domain Filter(s): %s", dnsname)
+				endpoints = append(endpoints[0:i], endpoints[i+1:]...)
+			} else {
```

Comment:
instead of having another else clause maybe leave and use continue?

```go
...
if dnsname == zone.Name || strings.HasSuffix(dnsname, "."+zone.Name) {
				// "pop" endpoint if it's matched to a residual zone... essentially a no-op
				log.Debugf("Ignoring Endpoint because it was matched to a zone that was not specified within Domain Filter(s): %s", dnsname)
				endpoints = append(endpoints[0:i], endpoints[i+1:]...)
			} 
continue
...

Reply:
@njuettner Thanks for the review! I agree that that would be cleaner in terms of code. I actually completed omitted this code stanza at first and changed the warning msg in line 382 to just be more encompassing in its description.

However, one thing that was asked for in the feature request was to suppress the:
External-DNS is logging warning"Unable to find company.org zone", every [interval (10s)].

which this logic takes in to account. "Popping" the endpoints off the slice allows the existing logic of having a warning below where len(endpoints) > 0, if no matching zones exist to continue to work.

If you think this is contributing too much of a code smell, I can see if I can do a refactor of this part of the code.